lib/Dashboard/TalkWidget.php

```diff
@@ -84,4 +124,89 @@ public function load(): void {
 		Util::addStyle('spreed', 'icons');
 		Util::addScript('spreed', 'talk-dashboard');
 	}
+
+	public function getItems(string $userId, ?string $since = null, int $limit = 7): array {
+		$rooms = $this->manager->getRoomsForUser($userId, [], true);
+
+		$rooms = array_filter($rooms, static function (Room $room) use ($userId) {
+			$participant = $room->getParticipant($userId);
+			$attendee = $participant->getAttendee();
+			return $room->getLastMessage() && $room->getLastMessage()->getId() > $attendee->getLastReadMessage();
+		});
+
+		uasort($rooms, [$this, 'sortRooms']);
+
+		$rooms = array_slice($rooms, 0, $limit);
+
+		$result = [];
+		foreach ($rooms as $room) {
+			$result[] = $this->prepareRoom($room, $userId);
+		}
+
+		return $result;
+	}
+
+	protected function prepareRoom(Room $room, string $userId): WidgetItem {
+		$participant = $room->getParticipant($userId);
+		$subtitle = '';
+
+		$lastMessage = $room->getLastMessage();
+		if ($lastMessage instanceof IComment) {
+			$message = $this->messageParser->createMessage($room, $participant, $room->getLastMessage(), $this->l10n);
+			$this->messageParser->parseMessage($message);
+			if ($message->getVisibility()) {
+				$placeholders = $replacements = [];
+
+				foreach ($message->getMessageParameters() as $placeholder => $parameter) {
+					$placeholders[] = '{' . $placeholder . '}';
+					if ($parameter['type'] === 'user' || $parameter['type'] === 'guest') {
+						$replacements[] = '@' . $parameter['name'];
+					} else {
+						$replacements[] = $parameter['name'];
+					}
+				}
+
+				$subtitle = str_replace($placeholders, $replacements, $message->getMessage());
+			}
+		}
+
+		return new WidgetItem(
+			$room->getDisplayName($userId),
+			$subtitle,
+			$this->url->linkToRouteAbsolute('spreed.Page.showCall', ['token' => $room->getToken()]),
+			$this->getRoomIconUrl($room, $userId)
+		);
```

Comment:
Can you add here roundIcon: true

https://github.com/nextcloud/server/blob/master/lib/public/Dashboard/Model/WidgetOptions.php#L32
it is in
Widget Options:
https://github.com/nextcloud/server/blob/master/lib/public/Dashboard/IOptionWidget.php#L37

Reply:
Items itself don't have the options and the widget itself already has the "roundIcon" option set.
https://github.com/nextcloud/spreed/pull/7942/files#diff-9d0ac71d126947c6f952b00aba3afc1df1e75b99431cb53086b0c43c2f677c29R90